libopenage/util/fixed_point.h

```diff
@@ -446,8 +447,37 @@ class FixedPoint {
 		return is;
 	}
 
+	/**
+	 * Pure FixedPoint sqrt implementation using Heron's Algorithm.
+	 *
+	 * Note that this function is undefined for negative values.
+	 */
 	constexpr double sqrt() {
-		return std::sqrt(this->to_double());
+		// Zero can cause issues later, so deal with now.
+		if (this->raw_value == 0) {
+			return 0.0;
+		}
+
+		// A greater shift = more precision, but can overflow the intermediate type if too large.
+		size_t max_shift = std::countl_zero((unsigned_intermediate_type)this->raw_value) - 1;
```

Comment:
use `static_cast` instead of C casts here to convert to a different type.

```suggestion
		size_t max_shift = std::countl_zero(static_cast<unsigned_intermediate_type>(this->raw_value)) - 1;
```